torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2027,8 +2034,14 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
   // Enqueue P2P op so that it can be cancelled by NCCL watchdog
   c10::cuda::CaptureStatus capture_status =
       c10::cuda::currentStreamCaptureStatusMayInitCtx();
+
+  // Notify graphs before we check the capture status preemptively
+  at::cuda::CUDAGraph::inc_pending_event_queries();
+
   if (!coalescing_state_ && capture_status == c10::cuda::CaptureStatus::None) {
     workEnqueue(work);
+  } else {
+    at::cuda::CUDAGraph::dec_pending_event_queries();
```

Comment:
Same here.

Reply:
See comment below on this---it's not the prettiest but I believe it's safe to increment before we check---otherwise it's potentially too late to notify the graphs-side code.